File: data_struct/Stack.cpp
#include <iostream>


class Stack{
public:
    int len;
    int top = -1;
    char* arr;

    explicit Stack(int lenght){
        len = lenght;
        arr = new char[len];
    }

    void push(char a){
        top++;
        arr[top] = a;
    }

    char pop(){
        top--;
        return arr[top];
    }

    char get_top(){
        return  arr[top];
    }

    bool empty(){
        return (top == -1);
    }

    void print_stack(){
        for(int i = 0; i < len; i++){
            std::cout << arr[i];
        }
        std::cout << std::endl;
    }



};

File: reverse_polish_notation_py/Stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, item):
        self.stack.append(item)

    def pop(self):
        if(self.isEmpty() == 0):
            return None
        return self.stack.pop()

    def isEmpty(self):
        return len(self.stack)

    def get_top(self):
        return self.stack[-1]